pandas/core/reshape/melt.py

```diff
@@ -45,6 +52,12 @@ def melt(frame, id_vars=None, value_vars=None, var_name=None,
                              ' columns are a MultiIndex')
         else:
             value_vars = list(value_vars)
+            # Check that `value_vars` are in frame
+            missing = Index(value_vars).difference(frame.columns)
+            if not missing.empty:
```

Comment:
This is a bit tricky since there are other checks on `id_vars` and `value_vars` independently, like these
https://github.com/pandas-dev/pandas/blob/c75ab23bdf237a9966a28f0b6941dea3ea5e0398/pandas/core/reshape/melt.py#L33
https://github.com/pandas-dev/pandas/blob/c75ab23bdf237a9966a28f0b6941dea3ea5e0398/pandas/core/reshape/melt.py#L51
So checking that all columns are present before these individual checks (like at L27) fails the test that addresses the above check.

Trying to check the columns after these checks (like at L64) works, but throws this warning:
`Passing list-likes to .loc or [] with any missing label will raise
KeyError in the future, you can use .reindex() as an alternative.`

Because of this line:
https://github.com/pandas-dev/pandas/blob/c75ab23bdf237a9966a28f0b6941dea3ea5e0398/pandas/core/reshape/melt.py#L61

Which is what I'm trying to avoid. 

Since the check needs to happen before L61 (to prevent the warning) and after the Multiindex checks (L33 and 51) I think doing them separately may be the easiest.

Reply:
ok that's fine. something failing, so merge master and ping on green.